src/tlo/util.py

```diff
@@ -498,15 +513,21 @@ def clean_dataframe(dataframes_dict: dict[str, DataFrame]) -> None:
         for _key, dataframe in dataframes_dict.items():
             all_data[_key] = dataframe.drop(dataframe.filter(like='Unnamed'), axis=1)  # filter and drop Unnamed columns
 
-    if files is None:
-        for f_name in folder.rglob("*.csv"):
-            all_data[f_name.stem] = pd.read_csv(f_name)
-
+    return_dict = False # a flag that will determine whether the output should be a dictionary or a DatFrame
+    if isinstance(files, list):
+        return_dict = True
+    elif files is None:
+        files = [f_name.stem for f_name in folder.glob("*.csv")]
+        return_dict = True
+    elif isinstance(files, str):
+        files = [files]
     else:
```

Comment:
At the moment this `else` clause will be run whenever `files` is not a instance `list` or `str` and is not `None`. Looking at the type hint / default value I guess this is meant to cover the case when `files` is an instance of an `int`? If so it would be better to make this `elif isinstance(files, int):` and add a separate `else:` clause that raises an error saying `files` is not of an expected type if none of the previous clauses have run. At the moment for example if a user passes a non-list iterable data structure as the `files` argument - for example a tuple or generator (as returned for example by `Path.glob` and `Path.rglob`) then the value passed will be silently overwritten which is probably not what they would expect. Failing loudly if `files` is of an unexpected type (and/or allowing `files` to be of more general iterable types by for example checking if it is an instance of `Iterable` from `collections.abc` instead of `list`) would be better here.

Reply:
You could also avoid the duplication of the list comprehension `[f_name.stem for f_name in folder.glob("*.csv")]` by having something like

```Python
elif isinstance(files, int) or files is None:
    files = [f_name.stem for f_name in folder.glob("*.csv")]
    return_dict = files is None
else:
    raise TypeError(f"Value passed for files argument {files} is not one of expected types.")
```

Reply:
The thought of combining those two conditions crossed my mind but I was lacking the smart way of setting `return_dict` to either True or False. thanks for this @matt-graham

Reply:
I believe using `isinstance(files, collections.abc.Iterable)` is not the best approach here, as it also considers strings to be iterable. If a string is included, it could alter the expected behavior of the return type when modelers specify a file name by passing a string. This would cause the method to return a dictionary instead of a DataFrame, which is likely not the intended outcome.

Reply:
Hi @mnjowe, good point around `str` being an instance of `Iterable`. We could sidestep that issue by having the `elif isinstance(files, str):` clause before the `elif isinstance(files, Iterable):` clause (as this would mean no string `files` reach the latter clause) but the current solution is fine here, the main thing was making sure we get an informative error message when we use a non-list iterable rather than the function doing something unexpected.

Reply:
Ah great! thanks @matt-graham . will definitely consider adding this the next time I'm to updating this method